frontend/apps/reader/modules/readerfooter.lua

```diff
@@ -1929,6 +1931,18 @@ function ReaderFooter:genAlignmentMenuItems(value)
     }
 end
 
+function ReaderFooter:addAdditionalFooterContent(value)
+    table.insert(self.external_footer_content, value)
+end
+
+function ReaderFooter:removeAdditionalFooterContent(pattern)
+    for i, v in pairs(self.external_footer_content) do
+        if v:find(pattern) then
+            table.remove(self.external_footer_content, i)
```

Comment:
This is a bit ugly.
Instead of adding text strings, and trying with luck to remove them - and instead of each time removing/adding the text, you could just add once (on :init() or something similar early) a function object, that footer/coptlistener would just _call_ to get the text. And that function could just return some `self.current_text`.
With a function object, you could just remove it by checking for equality.
And you would not need to broadcast anything to get the text update: the call would get you the most up to date value.